presto-main/src/main/java/com/facebook/presto/operator/UncheckedByteArrays.java

```diff
@@ -0,0 +1,132 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+import static java.util.Objects.requireNonNull;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_BYTE_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_SHORT_INDEX_SCALE;
+
+public class UncheckedByteArrays
+{
+    private static final Unsafe unsafe;
+
+    static {
+        try {
+            Field field = Unsafe.class.getDeclaredField("theUnsafe");
+            field.setAccessible(true);
+            unsafe = (Unsafe) field.get(null);
+            if (unsafe == null) {
+                throw new RuntimeException("Unsafe access not available");
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private UncheckedByteArrays()
+    {}
+
+    public static byte getByte(byte[] bytes, int index)
+    {
+        return unsafe.getByte(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int setByte(byte[] bytes, int index, byte value)
+    {
+        unsafe.putByte(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, value);
+        return index + ARRAY_BYTE_INDEX_SCALE;
+    }
+
+    public static short getShort(byte[] bytes, int index)
+    {
+        return unsafe.getShort(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int setShort(byte[] bytes, int index, short value)
+    {
+        unsafe.putShort(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, value);
+        return index + ARRAY_SHORT_INDEX_SCALE;
+    }
+
+    public static int getInt(byte[] bytes, int index)
+    {
+        return unsafe.getInt(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int setInt(byte[] bytes, int index, int value)
+    {
+        unsafe.putInt(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, value);
+        return index + ARRAY_INT_INDEX_SCALE;
+    }
+
+    public static long getLong(byte[] bytes, int index)
+    {
+        return unsafe.getLong(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int setLong(byte[] bytes, int index, long value)
+    {
+        unsafe.putLong(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, value);
+        return index + ARRAY_LONG_INDEX_SCALE;
+    }
+
+    public static int fill(byte[] bytes, int index, int length, byte value)
+    {
+        unsafe.setMemory(bytes, index + ARRAY_BYTE_BASE_OFFSET, length, value);
+        return index + length;
+    }
+
+    public static int setBytes(byte[] bytes, int index, byte[] values, int offset, int length)
```

Comment:
@mbasmanova setBytes calls unsafe.copyMemory() and setInts calls unsafe.putInt(). I don't quite understand why you said it's not unsafe? If we want to move these two methods out, we might also want to move fill() out because these three methods were not from the original airlift ByteArrays class and is copying/setting more than just one value like other methods in this class.

Reply:
@yingsu00 I should have been clearer in my comment. Most methods in `UncheckedByteArrays` are copy-paste of the same methods from `ByteArrays`, but they omit range check which makes them "unchecked" and not super safe to use. However, `setBytes` and `setInts` seem to be perfectly safe to use and actually include range checks. Hence, they belong to `ByteArrays`, not `UncheckedByteArrays`. Since changing airlift is hard, we should put these into a new class. I'd name it `MoreByteArrays`. As you pointed out, `fill` method should go these too.

Also, I noticed that `setXxx` methods from `UncheckedByteArrays` are used with static imports. Hence, when reading the code it is not clear that these are "be careful with" methods. Let's rename these methods to add "Unchecked" suffix, e.g. `setByte` -> `setByteUnchecked` (similar to `UncheckedBlock` methods.)